src/opm/input/eclipse/Schedule/MSW/Segment.cpp

```diff
@@ -116,81 +115,118 @@ namespace {
+    Segment::Segment(const Segment& src,
+                     const double   new_depth,
+                     const double   new_length,
+                     const double   new_volume)
+        : Segment { src, new_depth, new_length }
+    {
+        this->m_volume = new_volume;
+    }
 
+    Segment::Segment(const Segment& src,
+                     const double   new_depth,
+                     const double   new_length)
+        : Segment(src)
+    {
+        this->m_depth = new_depth;
+        this->m_total_length = new_length;
+        this->m_data_ready = true;
+    }
 
-    Segment::Segment(int segment_number_in, int branch_in, int outlet_segment_in, double length_in, double depth_in,
-                     double internal_diameter_in, double roughness_in, double cross_area_in,
-                     double volume_in, bool data_ready_in)
-    : m_segment_number(segment_number_in),
-      m_branch(branch_in),
-      m_outlet_segment(outlet_segment_in),
-      m_total_length(length_in),
-      m_depth(depth_in),
-      m_internal_diameter(internal_diameter_in),
-      m_roughness(roughness_in),
-      m_cross_area(cross_area_in),
-      m_volume(volume_in),
-      m_data_ready(data_ready_in)
+    Segment::Segment(const Segment& src, const double new_volume)
+        : Segment(src)
     {
+        this->m_volume = new_volume;
     }
 
-  Segment::Segment(const Segment& src, double new_depth, double new_length):
-      Segment(src)
-  {
-      this->m_depth = new_depth;
-      this->m_total_length = new_length;
-      this->m_data_ready = true;
-  }
-
-  Segment::Segment(const Segment& src, double new_depth, double new_length, double new_volume):
-    Segment(src, new_depth, new_length)
-   {
-       this->m_volume = new_volume;
-   }
-
-  Segment::Segment(const Segment& src, double new_volume):
-      Segment(src)
-  {
-      this->m_volume = new_volume;
-  }
-
-  Segment Segment::serializeObject()
-  {
-      Segment result;
-      result.m_segment_number = 1;
-      result.m_branch = 2;
-      result.m_outlet_segment = 3;
-      result.m_inlet_segments = {4, 5};
-      result.m_total_length = 6.0;
-      result.m_depth = 7.0;
-      result.m_internal_diameter = 8.0;
-      result.m_roughness = 9.0;
-      result.m_cross_area = 10.0;
-      result.m_volume = 11.0;
-      result.m_data_ready = true;
-      result.m_icd = SICD::serializeObject();
-      return result;
-  }
+    Segment Segment::serializeObject()
+    {
+        Segment result;
+        result.m_segment_number = 1;
+        result.m_branch = 2;
+        result.m_outlet_segment = 3;
+        result.m_inlet_segments = {4, 5};
+        result.m_total_length = 6.0;
+        result.m_depth = 7.0;
+        result.m_internal_diameter = 8.0;
+        result.m_roughness = 9.0;
+        result.m_cross_area = 10.0;
+        result.m_volume = 11.0;
+        result.m_data_ready = true;
+        result.m_icd = SICD::serializeObject();
+        result.m_x = 12.0;
+        result.m_y = 13.0;
```

Comment:
I am probably still not getting the serializeObject function, but I always assumed when reading from the stream it should overwrite he values set above. In that case shouldn't `SICD::serializeObject()` be the last call before returning the result?